### DISCUSSION TITLE
Deploy > managed-learn > ml-core-service

### DISCUSSION THREAD

**Discussion by sandeep-r1** [Sunbird-Ed/Community#530] [Bugs]
Hi Team,
We are facing issue while deploying ml-core-service. Please find the error below. Kindly help us to resolve this issue

```
TASK [sunbird-deploy : helm install] *******************************************
fatal: [localhost]: FAILED! => {
    "changed": true,
    "cmd": "helm upgrade --install --atomic --timeout 10m ml-core-service /var/lib/jenkins/workspace/Deploy/alex/dev/managed-learn/ml-core-service/kubernetes/helm_charts/core/ml-core-service -n alexdev",
    "delta": "0:10:00.684825",
    "end": "2023-05-30 15:28:13.507233",
    "rc": 1,
    "start": "2023-05-30 15:18:12.822408"
}

STDOUT:

Release "ml-core-service" does not exist. Installing it now.


STDERR:

Error: release ml-core-service failed, and has been uninstalled due to atomic being set: timed out waiting for the condition


MSG:

non-zero return code


PLAY RECAP *********************************************************************
localhost                  : ok=16   changed=14   unreachable=0    failed=1    skipped=5    rescued=0    ignored=0 
```

**Comment by tsprasath** [comment]
@sandeep-r1  Please share the pods logs or the events of the deployment

**Reply by srikanth-6520** [reply]
@tsprasath here we have provided pod logs of the service.

 APPLICATION_PORT         │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ APPLICATION_ENV          │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ MONGODB_URL              │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ INTERNAL_ACCESS_TOKEN    │ Required internal access token │
├──────────────────────────┼────────────────────────────────┤
│ CLOUD_STORAGE            │ Enable/Disable cloud services  │
├──────────────────────────┼────────────────────────────────┤
│ GCP_PATH                 │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ GCP_BUCKET_NAME          │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ AZURE_ACCOUNT_NAME       │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ AZURE_ACCOUNT_KEY        │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ AZURE_STORAGE_CONTAINER  │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ AWS_ACCESS_KEY_ID        │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ AWS_SECRET_ACCESS_KEY    │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ AWS_BUCKET_NAME          │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ AWS_BUCKET_REGION        │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ AWS_BUCKET_ENDPOINT      │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ KEYCLOAK_PUBLIC_KEY_PATH │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ ML_SURVEY_SERVICE_URL    │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ ML_PROJECT_SERVICE_URL   │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ USER_SERVICE_URL         │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ CSV_REPORTS_PATH         │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ APP_PORTAL_BASE_URL      │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ FORM_SERVICE_URL         │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ OCI_ACCESS_KEY_ID        │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ OCI_SECRET_ACCESS_KEY    │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ OCI_BUCKET_NAME          │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ OCI_BUCKET_REGION        │ PASSED                         │
├──────────────────────────┼────────────────────────────────┤
│ OCI_BUCKET_ENDPOINT      │ PASSED                         │
└──────────────────────────┴────────────────────────────────┘
Server could not start . Not all environment variable is provided

**Reply by tsprasath** [reply]
[aks30](https://github.com/aks30) Kindly help @srikanth-6520 on ML service

**Reply by aks30** [reply]
@tsprasath As the message indicates certain mandatory config which is required for the service to start is not being supplied, if you can share further, the list of all such keys against which the message is not. "PASSED", it would help identify where is it failing

**Reply by gandham-santhosh** [reply]
@srikanth-6520 @sandeep-r1 Please mark this thread as answered if the issue is resolved

**Comment by sandeep-r1** [comment]
Answered
